contracts/src/v0.8/dev/automation/upkeeps/MercuryRegistry.sol

```diff
@@ -0,0 +1,280 @@
+  struct Feed {
+    uint32 observationsTimestamp; // the timestamp of the most recent data assigned to this feed
+    int192 price; // the current price of the feed
+    int192 bid; // the current bid price of the feed
+    int192 ask; // the current ask price of the feed
+    string feedName; // the name of the feed
+    string feedId; // the id of the feed (hex encoded)
+    bool active; // true if the feed is being actively updated, otherwise false
+  }
+
+  // Report object obtained from off-chain Mercury server.
+  struct Report {
+    bytes32 feedId; // the feed Id of the report
+    uint32 observationsTimestamp; // the timestamp of when the data was observed
+    int192 price; // the median value of the OCR round
+    int192 bid; // the median bid of the OCR round
+    int192 ask; // the median ask if the OCR round
+    uint64 blocknumberUpperBound; // the highest block observed at the time the report was generated
+    bytes32 upperBlockhash; // the blockhash of the highest block observed
+    uint64 blocknumberLowerBound; // the lowest block observed at the time the report was generated
+    uint64 currentBlockTimestamp; // the timestamp of the highest block observed
+  }
+
+  event FeedUpdated(uint32 observationsTimestamp, int192 price, int192 bid, int192 ask, string feedId);
+
+  uint32 private constant MIN_GAS_FOR_PERFORM = 200_000;
+
+  string constant c_feedParamKey = "feedIdHex"; // for Mercury v0.2 - format by which feeds are identified
+  string constant c_timeParamKey = "blockNumber"; // for Mercury v0.2 - format by which feeds are filtered to be sufficiently recent
+  IVerifierProxy public s_verifier; // for Mercury v0.2 - verifies off-chain reports
+
+  int192 constant scale = 1_000_000; // a scalar used for measuring deviation with precision
+  int192 public s_deviationPercentagePPM; // acceptable deviatoin threshold - 1.5% = 15_000, 100% = 1_000_000, etc..
+  uint32 public s_stalenessSeconds; // acceptable staleness threshold - 60 = 1 minute, 300 = 5 minutes, etc..
+
+  string[] public s_feeds; // list of feed Ids
+  mapping(string => Feed) public s_feedMapping; // mapping of feed Ids to stored feed data
+
+  constructor(
+    string[] memory feedIds,
+    string[] memory feedNames,
+    address verifier,
+    int192 deviationPercentagePPM,
+    uint32 stalenessSeconds
+  ) ConfirmedOwner(msg.sender) {
+    s_verifier = IVerifierProxy(verifier);
+
+    // Store desired deviation threshold and staleness seconds.
+    s_deviationPercentagePPM = deviationPercentagePPM;
+    s_stalenessSeconds = stalenessSeconds;
+
+    // Store desired feeds.
+    setFeeds(feedIds, feedNames);
+  }
+
+  // Returns a user-defined batch of feed data, based on the on-chain state.
+  function getLatestFeedData(string[] memory feedIds) external view returns (Feed[] memory) {
+    Feed[] memory feeds = new Feed[](feedIds.length);
+    for (uint256 i = 0; i < feedIds.length; i++) {
+      feeds[i] = s_feedMapping[feedIds[i]];
```

Comment:
I guess no checking of whether this feed ID exists in the mapping - user just needs to check return data?

Reply:
If the FeedID is not in the mapping, `active` will be false. There may still be data from prior updates, but they will ultimately become stale.